src/main/java/uk/gov/pay/ledger/transaction/dao/TransactionDao.java

```diff
@@ -20,7 +21,9 @@
 
     private static final String FIND_TRANSACTION_BY_EXTERNAL_ID_AND_GATEWAY_ACCOUNT_ID = "SELECT * FROM transaction " +
             "WHERE external_id = :externalId " +
-            "and gateway_account_id = :gatewayAccountId";
+            "AND gateway_account_id = :gatewayAccountId " +
+            "AND (:transactionType::transaction_type is NULL OR type = :transactionType::transaction_type) " +
+            "AND (:parentExternalId is NULL OR parent_external_id = :parentExternalId)";
```

Comment:
Are we making it mandatory to provide `parentExternalId` for get one refund transaction ?

Reply:
I don't think it needs to be mandatory on ledger side. If it's not valid, it won't return any results.

Reply:
This could break if `:parentExternalId` is an empty string which could be ignored ?

Reply:
Should it be an empty string?

Reply:
This is the usual pattern for non-mandatory fields until it is different in this case 

https://github.com/alphagov/pay-ledger/blob/bc62be837a511359e961dcad08293e4c6947a943/src/main/java/uk/gov/pay/ledger/transaction/search/common/TransactionSearchParams.java#L271

Reply:
I don't think it's required in this case. Public api doesn't specify this parameter at all when calling ledger, so it shouldn't be an issue.
If we discover otherwise we can update the code to handle the scenario.